### PR TITLE
move changes from previous branche

### PR DESCRIPTION
**Pull request by Tiana1203**

### CODE REVIEW

File: API/API/DataAccessLayer/Models/User.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace API.DataAccessLayer.Models
+{
+    public class User
+    {
+        public int Id { get; set; }
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Models/User.cs]: id should have Guid type

File: API/API/DataAccessLayer/Models/User.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+        public string Email { get; set; }
+
+        public string Password { get; set; }
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Models/User.cs]: passwordHash

File: API/API/DataAccessLayer/Repositories/IRepository.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿namespace API.DataAccessLayer.Repositories
+{
+    public interface IRepository<T> where T : class
+    {
+        Task<List<T>> GetAll();
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Repositories/IRepository.cs]: add async word to asynchronous method names. check this everywhere

File: API/API/DataAccessLayer/ShopdbContext.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        string password = "";
+
+        public ShopdbContext(DbContextOptions<ShopdbContext> options)
+            : base(options)
+        {
+            Database.EnsureCreated();
+        }
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        => optionsBuilder
+            .UseMySql($"host=localhost; database=shopdb; port=3306; username={login}; password={password}", new MySqlServerVersion(new Version(8, 0, 35)));
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/ShopdbContext.cs]: get this information from appsettings.json

File: API/API/Program.cs [hunk 5]
```diff
@@ -1,8 +1,22 @@
...
+var configBuilder = new ConfigurationBuilder();
+configBuilder.SetBasePath(Directory.GetCurrentDirectory());
+configBuilder.AddJsonFile("appsettings.json");
+var config = configBuilder.Build();
+var connectionString = config.GetConnectionString("DefaultConnection");
+
+builder.Services.AddDbContext<ShopdbContext>(options => options
+.UseMySql(connectionString, new MySqlServerVersion(new Version(8, 0, 35))));
+
+builder.Services.AddTransient<IRepository<User>, UserRepository>();
```
Comment (SUPERKISA8BIIT) [inline API/API/Program.cs]: to the extension method

Comment (Tiana1203) [reply API/API/Program.cs]: What to move to another method?

Comment (SUPERKISA8BIIT) [reply API/API/Program.cs]: line 8 - 15

Comment (SUPERKISA8BIIT) [reply API/API/Program.cs]: and line 17 - to another extension method

File: API/API/DataAccessLayer/Repositories/UserRepository.cs [hunk 6]
```diff
@@ -0,0 +1,65 @@
+﻿using API.DataAccessLayer.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.DataAccessLayer.Repositories
+{
+    public class UserRepository : IRepository<User>
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Repositories/UserRepository.cs]: you should define abstract class BaseRepository and implement all CRUD operations with database there. inherit from BaseRepository class in UserRepository

File: API/API/DataAccessLayer/Repositories/IRepository.cs [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+        Task<T?> Get(int id);
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Repositories/IRepository.cs]: use CansellationToken in asyncronious methods

File: API/API/DataAccessLayer/Repositories/IRepository.cs [hunk 8]
```diff
@@ -0,0 +1,11 @@
+﻿namespace API.DataAccessLayer.Repositories
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Repositories/IRepository.cs]: move this to interfaces folder and define specified interface for userRepository

File: API/API/DataAccessLayer/Repositories/UserRepository.cs [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        public async Task<int> Delete(int id)
+        {
+            int res = await _dbContext.Users.Where(u => u.Id == id).ExecuteDeleteAsync();
+            await _dbContext.SaveChangesAsync();
+            return res;
+        }
+
+        public async Task<User?> Get(int id)
+        {
+            var res = await _dbContext.Users.Where(u => u.Id == id).FirstOrDefaultAsync();
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Repositories/UserRepository.cs]: use .AsNoTracking() in reading methods

File: API/API/DataAccessLayer/Repositories/UserRepository.cs [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        public Task<List<User>> GetAll()
+        {
+            return _dbContext.Users.ToListAsync();
+        }
+
+        public async Task<User?> Update(User item)
+        {
+            var res = await _dbContext.Users.Where(u => u.Id == item.Id).FirstOrDefaultAsync();
+            User u;
+            if (res == null)
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Repositories/UserRepository.cs]: you should check if user exists in BLL layer, not in DAL

File: API/API/DataAccessLayer/Repositories/UserRepository.cs [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+            {
+                return res;
+            }
+            else
+            {
+                u = res;
+                if (item.Email != null)
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Repositories/UserRepository.cs]: this is also BLL logic

File: API/API/DataAccessLayer/Repositories/UserRepository.cs [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+
+        public Task<List<User>> GetAll()
+        {
+            return _dbContext.Users.ToListAsync();
+        }
+
+        public async Task<User?> Update(User item)
+        {
+            var res = await _dbContext.Users.Where(u => u.Id == item.Id).FirstOrDefaultAsync();
+            User u;
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Repositories/UserRepository.cs]: don't use one-letter names of variables

**Review by SUPERKISA8BIIT — COMMENTED** [review]
checked

**Comment by Tiana1203** [comment]
im nor sure about BaseRepository.UpdateAsync cause it will generate 2 query for db for remove and add. Correct me if im wrong.

**Pull request closed without merging**